src/kernel/notifications_interface.h

```diff
@@ -27,6 +28,12 @@ class Notifications
     virtual void headerTip(SynchronizationState state, int64_t height, int64_t timestamp, bool presync) {}
     virtual void progress(const bilingual_str& title, int progress_percent, bool resume_possible) {}
     virtual void warning(const bilingual_str& warning) {}
+
+    //! The fatal error notification is sent to notify the user and start
+    //! shutting down if an error happens in kernel code that can't be recovered
+    //! from. After this notification is sent, whatever function triggered the
+    //! error should also return an error code or raise an exception.
```

Comment:
Sorry for the late conceptual review, but I'm worried about how true this is and how well it's enforced? IIRC last time I played with my clang-tidy plugin, there were at least a few places where StartShutdown() was called (maybe in a nested function) but then the function didn't return immediately.

Reply:
I'm more worried about the enforcement question. I'm nearly positive that this assumption is not true now, and even if so there's nothing other than review keeping it true in the future.

I'll work on generating a more current shutdown-bubble-up demo to demonstrate the problem.